internal/api/dex/proxy.go

```diff
@@ -0,0 +1,71 @@
+package dex
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"os"
+
+	"github.com/kelseyhightower/envconfig"
+	"github.com/pkg/errors"
+)
+
+// ProxyConfig represents configuration for a reverse proxy to a Dex server.
+type ProxyConfig struct {
+	// ServerAddr is the address of the target Dex server, beginning with https://
+	ServerAddr string `envconfig:"DEX_SERVER_ADDRESS" required:"true"`
```

Comment:
Minor nit: is there a time when this is not `http://dex`? If so we might consider having a default that lessens user's configuration.

Reply:
Good question.

iirc, the self-signed cert that the chart generates by default (which you can opt out of if you really want to provide your own) is valid only for `<release name>-kargo-dex-server.<namespace>.svc`, so that's the host name that has to be used or else the cert won't be trusted.

I'll open a separate, low-priority issue to see if I can add `dex` as a SAN on that cert, in which case, we can then use `https://dex` as the default value for that field.